spring-beans/src/main/java/org/springframework/beans/BeanUtils.java

```diff
@@ -710,17 +713,25 @@ private static void copyProperties(Object source, Object target, @Nullable Class
 			if (writeMethod != null && (ignoreList == null || !ignoreList.contains(targetPd.getName()))) {
 				PropertyDescriptor sourcePd = getPropertyDescriptor(source.getClass(), targetPd.getName());
 				if (sourcePd != null) {
+					Field sourcefield = ReflectionUtils.findField(source.getClass(), sourcePd.getName());
+					Field targetfield = ReflectionUtils.findField(target.getClass(), targetPd.getName());
```

Comment:
There is no requirement that the components have fields with those names. Thus, this approach (i.e., working with the fields) can potentially lead to null pointer exceptions.

Reply:
yeah got that